doc/source/whatsnew/v1.2.4.rst

```diff
@@ -20,7 +20,7 @@ Fixed regressions
 - Fixed regression in (in)equality comparison of ``pd.NaT`` with a non-datetimelike numpy array returning a scalar instead of an array (:issue:`40722`)
 - Fixed regression in :meth:`DataFrame.where` not returning a copy in the case of an all True condition (:issue:`39595`)
 - Fixed regression in :meth:`DataFrame.replace` raising ``IndexError`` when ``regex`` was a multi-key dictionary (:issue:`39338`)
--
+- Fixed regression in repr of floats in an ``object`` column not respecting ``float_format`` (:issue:`40024`)
```

Comment:
is this *only* for to_html? or more generally?

Reply:
I guess *also* console repr (so mention these 2). doesn't affect, csv for example, right?

Reply:
Yep it affects to_html and console repr. 

In theory we might want this to affect `to_csv`, but this doesn't change behavior right now because formatting is handled based on column dtype, such that `float_format` doesn't get used for object dtype columns even if some values in that column are floats.

Reply:
Also affects `to_latex`, but not `to_json`

Reply:
ok pls list those and add a test for to_latex if you can

Reply:
Have updated whatsnew and added to_latex test